WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -126,6 +141,53 @@ struct ThemesPreviewView: View {
             }
             .navigationBarTitleDisplayMode(.inline)
         }
+        .task { await viewModel.fetchPages() }
+        .sheet(isPresented: $showPagesMenu) {
+            pagesListSheet(pages: viewModel.pages)
+        }
+    }
+
+    @ViewBuilder
+    private var pageSelector: some View {
+
+        // Here we show only a "Preview" label with no selector, both for loading and error cases.
+        // In the case of page loading error, the home page is still usable, so showing "Preview" is better than nothing.
+        switch viewModel.state {
+        case .pagesLoading, .pagesLoadingError:
+            Text(Localization.preview)
+
+        case .pagesContent:
+            Button(action: { showPagesMenu = true }) {
+                HStack {
+                    Text(viewModel.selectedPage.title).bodyStyle()
+                    Image(uiImage: .chevronDownImage)
+                        .fixedSize()
+                        .bodyStyle()
+                }
+            }
+        }
+    }
+
+    private func pagesListSheet(pages: [WordPressPage]) -> some View {
+        return VStack {
+            Text(Localization.pagesSheetHeading)
+                .subheadlineStyle()
+                .padding(Layout.pagesSheetPadding)
+            ForEach(pages) { page in
+                Button(action: {
+                    viewModel.setSelectedPage(page: page)
+                    showPagesMenu = false
+                }, label: {
+                    Text(page.title)
+                        .bodyStyle()
+                        .frame(maxWidth: .infinity, alignment: .leading)
+
+                })
+                .padding(Layout.contentPadding)
+            }
+
+            Spacer()
+        }
```

Comment:
Opinion:

1. Make this a `var` to keep it consistent with `pageSelector` above. 
1. Remove the `return` statement to improve readability.
1. Remove parameter and read from view model directly. 

```suggestion
    private var pagesListSheet: some View {
        VStack {
            Text(Localization.pagesSheetHeading)
                .subheadlineStyle()
                .padding(Layout.pagesSheetPadding)
            ForEach(viewModel.pages) { page in
                Button(action: {
                    viewModel.setSelectedPage(page: page)
                    showPagesMenu = false
                }, label: {
                    Text(page.title)
                        .bodyStyle()
                        .frame(maxWidth: .infinity, alignment: .leading)
                })
                .padding(Layout.contentPadding)
            }

            Spacer()
        }
```

Reply:
Thanks! this makes sense.